test/test_nestedtensor.py

```diff
@@ -272,14 +273,52 @@ def test_to_padded_tensor_simple(self, device):
 
             self.assertEqual(padded, correct_output)
             self.assertEqual(padded.device, torch.device(device))
+            self.assertEqual(padded.dtype, dtype)
+
+    @dtypes(torch.float, torch.float16)
+    def test_to_padded_tensor_dim2(self, device, dtype):
+        ts = [
+            torch.randn(1, device=device, dtype=dtype),
+            torch.randn(2, device=device, dtype=dtype),
+            torch.randn(4, device=device, dtype=dtype),
+        ]
+        nt = torch.nested_tensor(ts, device=device, dtype=dtype)
+        pad = 42
+        correct_output = []
+        for t in ts:
+            next_output = torch.ones_like(ts[2]) * pad
+            correct_output.append(next_output)
+            next_output[:t.size(0)].copy_(t)
+        correct_output = torch.stack(correct_output)
+        padded = nt.to_padded_tensor(pad)
+        self.assertEqual(padded, correct_output)
 
-    def test_to_padded_tensor_unrelated_shapes(self, device):
+    @dtypes(torch.float, torch.float16)
+    def test_to_padded_tensor_dim3(self, device, dtype):
         ts = [
-            torch.randn(1, 2, 3, device=device),
-            torch.randn(2, 3, 4, device=device),
-            torch.randn(4, 5, 6, device=device),
+            torch.randn(1, 2, device=device, dtype=dtype),
```

Comment:
you might want to test bigger inputs, so far the "vectorized" part of your loops are never tested.